Brokerages/Samco/SamcoBrokerage.cs

```diff
@@ -0,0 +1,1084 @@
+        /// </summary>
+        /// <param name="tradingSegment">Trading Segment</param>
+        /// <param name="productType">Product Type</param>
+        /// <param name="apiKey">api key</param>
+        /// <param name="apiSecret">api secret</param>
+        /// <param name="algorithm">the algorithm instance is required to retrieve account type</param>
+        /// <param name="yob">year of birth</param>
+        public SamcoBrokerage(string tradingSegment, string productType, string apiKey, string apiSecret, string yob, IAlgorithm algorithm, IDataAggregator aggregator)
+            : base("Samco")
+        {
+            _tradingSegment = tradingSegment;
+            _samcoProductType = productType;
+            _algorithm = algorithm;
+            _securityProvider = algorithm.Portfolio;
+            _aggregator = aggregator;
+            _samcoAPI = new SamcoBrokerageAPI();
+            _samcoAPI.Authorize(apiKey, apiSecret, yob);
+            _symbolMapper = new SamcoSymbolMapper();
+            _messageHandler = new BrokerageConcurrentMessageHandler<WebSocketMessage>(OnMessageImpl);
+
+            var subscriptionManager = new EventBasedDataQueueHandlerSubscriptionManager();
+            _algorithm.SetOptionChainProvider(new SamcoLiveOptionChainProvider(_symbolMapper));
+
+            WebSocket = new SamcoWebSocketClientWrapper();
+            WebSocket.Initialize("");
+            WebSocket.SetAuthTokenHeader(_samcoAPI.token);
+            WebSocket.Message += OnMessage;
+            WebSocket.Open += (sender, args) =>
+            {
+                Log.Trace($"SamcoBrokerage(): WebSocket.Open. Subscribing");
+                Subscribe(GetSubscribed());
+            };
+            WebSocket.Error += OnError;
+
+            subscriptionManager.SubscribeImpl += (s, t) =>
+            {
+                Subscribe(s);
+                return true;
+            };
+            subscriptionManager.UnsubscribeImpl += (s, t) => Unsubscribe(s);
+
+            _subscriptionManager = subscriptionManager;
+            _fillMonitorTask = Task.Factory.StartNew(FillMonitorAction, _ctsFillMonitor.Token);
+            Log.Trace("Start Samco Brokerage");
+        }
+
+        private void FillMonitorAction()
+        {
+            Log.Trace("SamcoBrokerage.FillMonitorAction(): task started");
+
+            try
+            {
+                foreach (var order in GetOpenOrders())
+                {
+                    _pendingOrders.TryAdd(order.BrokerId.First(), order);
+                }
+
+                while (!_ctsFillMonitor.IsCancellationRequested)
+                {
+                    _fillMonitorResetEvent.WaitOne(TimeSpan.FromMilliseconds(_fillMonitorTimeout), _ctsFillMonitor.Token);
+
+                    foreach (var kvp in _pendingOrders)
+                    {
+                        var orderId = kvp.Key;
+                        var order = kvp.Value;
+
+                        var response = _samcoAPI.GetOrderDetails(orderId);
```

Comment:
Seems order events are not event driven, does Samco not have order events? Note Zeroodha does